### DISCUSSION TITLE
How can I enable fetch before push but disable auto fetch?

### DISCUSSION THREAD

**Discussion by linsui** [jesseduffield/lazygit#3774] [Q&A] [answered]
I need to fetch the remote before push but I don't want it keep fetching in background. I guess I can set the fetchInterval to a large value. Is there a better solution?

**Comment by mark2185** [comment]
Maybe adding `git fetch` as a `pre-push` hook?

**Reply by linsui** [reply]
I thought this needs to be added to every repo...

**Reply by mark2185** [reply, marked as answer]
You _could_ add a `core.hooksPath` and have a global hook there ([docs](https://github.com/git/git/blob/6675f501f6b987dbdb0dbeb1d2efeb5a27fc41a7/Documentation/config.txt#L621-L636)).

Second option would be to create a [custom command](https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Command_Keybindings.md) within `lazygit` and map it to `P` (push), something like `git fetch && git push`.

Third option would be to create a habit of fetching/pulling before pushing since it's something one should use even when not using any git wrappers.

**Reply by linsui** [reply]
Thanks!
